accounts-journey/presentation/viewmodel/AccountsListViewModel.swift

```diff
@@ -8,52 +8,151 @@
 import Foundation
 import Resolver
 import Combine
+import BackbaseDesignSystem
 
-final class AccountsListViewModel {
+enum AccountListEvent {
+    case getAccounts
+    case refresh
+    case search(String)
+}
+
+extension AccountListEvent: Equatable {}
+
+final class AccountsListViewModel: NSObject, ObservableObject {
     
-    enum Input {
-        case viewDidAppear
-        case viewDidRefresh
+    enum ScreenState {
+        case idle
+        case loading
+        case loaded
+        case emptyResults(StateViewConfiguration)
+        case hasError(StateViewConfiguration)
     }
     
-    enum Output {
-        case fetchDidFail(error: Error)
-        case fetchDidSucceed(accountSummary: AccountsJourney.AccountsSummary)
+    @Published var allAccounts = [AccountUiModel]()
+    private var cancellables = Set<AnyCancellable>()
+    
+    @Published var hasError = false
+    var filteredAccounts = [AccountUiModel]()
+    
+    private let screenStateSubject = CurrentValueSubject<ScreenState, Never>(.idle)
+    
+    var screenStatePublisher: AnyPublisher<ScreenState, Never> {
```

Comment:
I think the screenStatePublisher and the screenStateSubject can be just one @Published var.